### PR TITLE
codesmells and coverage

### PR DESCRIPTION
**Pull request by nithyal1**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java [hunk 1]
```diff
@@ -243,15 +257,17 @@ public IAMServiceAccountSecret rotateIAMSecret(IAMServiceAccountRotateRequest ia
 
         StringBuilder jsonResponse = new StringBuilder();
 
+        BufferedReader r = null;
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java:260]: This code is not needed and it is fixed for SonarQube bugs fix

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java [hunk 2]
```diff
@@ -243,15 +257,17 @@ public IAMServiceAccountSecret rotateIAMSecret(IAMServiceAccountRotateRequest ia
...
+        BufferedReader br = null;
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java:261]: This code is not needed and it is fixed for SonarQube bugs fix

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java [hunk 3]
```diff
@@ -273,10 +289,25 @@ public IAMServiceAccountSecret rotateIAMSecret(IAMServiceAccountRotateRequest ia
         } catch (IOException e) {
             log.error(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
                     put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER)).
-                    put(LogMessage.ACTION, "rotateIAMSecret").
+                    put(LogMessage.ACTION, IAMServiceAccountConstants.ROTATE_IAM_SECRET).
                     put(LogMessage.MESSAGE, "Failed to parse IAM Secret response").
                     put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL)).
                     build()));
+        }finally {
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java:296]: This code is not needed and it is fixed for SonarQube bugs fix

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java [hunk 4]
```diff
@@ -502,8 +532,8 @@ public Response updateIAMSvcAccNewAccessKeyIdInMetadata(String token, String aws
                             build()));
                 }
 
-                String writeJson =  "{\"path\":\""+path+"\",\"data\":"+ metadataJson +"}";
-                metadataResponse = requestProcessor.process("/write",writeJson,token);
+                String writeJson =  path+path+data+ metadataJson +"}";
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java:535]: Please double check... why the "path" variable appending twice ?

**Review by alexrajan3 — COMMENTED** [review]
Please check the review comments below

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/model/IAMServiceAccountAWSRole.java [hunk 5]
```diff
@@ -134,22 +134,22 @@ public boolean equals(Object obj) {
 			if (other.access != null)
 				return false;
 		} else if (!access.equals(other.access))
-			return false;
+				return false;
```
[the remark this answers is not in the document]
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/model/IAMServiceAccountAWSRole.java:137]: adding indentation to fix codesmell

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/model/IAMServiceAccountNode.java [hunk 6]
```diff
@@ -63,18 +63,18 @@ public boolean equals(Object obj) {
 		if (folders == null) {
 			if (other.folders != null)
 				return false;
-		} else if (!folders.equals(other.folders))
-			return false;
+		} 	else if (!folders.equals(other.folders))
```
[the remark this answers is not in the document]
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/model/IAMServiceAccountNode.java:66]: adding indentation to fix code smell

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java (hunk 4 above)
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/utils/IAMServiceAccountUtils.java:535]: changed

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 7]
```diff
@@ -523,7 +527,7 @@ private boolean populateCertOwnerEmaild(SSLCertificateRequest sslCertificateRequ
                         if (targetSystemId == 0) {
                             enrollResponse.setResponse(SSLCertificateConstants.SSL_CREATE_EXCEPTION);
                             enrollResponse.setSuccess(Boolean.FALSE);
-                            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("{\"errors\":[\"" + enrollResponse.getResponse() + "\"]}");
+                            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("{\"errors\" :[\"" + enrollResponse.getResponse() + "\"]}");
```
[the remark this answers is not in the document]
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:530]: removed

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 8]
```diff
@@ -6582,12 +6588,14 @@ else if (TVaultConstants.OIDC.equals(vaultAuthMethod)) {
...
+
+            String metaDataPath = (certType.equalsIgnoreCase(SSLCertificateConstants.INTERNAL)) ?
                     SSLCertificateConstants.SSL_CERT_PATH + '/' + certificateName : SSLCertificateConstants.SSL_EXTERNAL_CERT_PATH + '/' + certificateName;
             //Permission path
-            String permissionMetaDataPath = (certType.equalsIgnoreCase("internal")) ?
+
+            String permissionMetaDataPath = (certType.equalsIgnoreCase(SSLCertificateConstants.INTERNAL)) ?
                     SSLCertificateConstants.SSL_CERT_PATH_VALUE + '/' + certificateName : SSLCertificateConstants.SSL_CERT_PATH_VALUE_EXT + '/' + certificateName;
-            Response response = getCertificateDetailsByMatadataPath(metaDataPath, authToken);
+            Response response = getCertificateDetailsByMatadataPath(metaDataPath, token);
```
[the remark this answers is not in the document]
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:6598]: changed to authToken

**Review by alexrajan3 — APPROVED** [review]

**Pull request merged**